packages/builtin-components/src/components/radar.js

```diff
@@ -3,52 +3,42 @@ import PropTypes from 'prop-types';
 import ReactEcharts from 'echarts-for-react';
 import _ from 'lodash';
 import { validate } from '../utils';
+import { timeInterval } from 'rxjs/operator/timeInterval';
 
 export default class Radar extends PureComponent {
   static propTypes = {
-    source: PropTypes.arrayOf(PropTypes.array).isRequired,
+    source: PropTypes.arrayOf(PropTypes.object).isRequired,
   }
 
+
+
   render() {
-    const { source } = this.props;
-    // transpose two-dimensional array
-    /**
-     *
-      const source = [
-        ['product', '2015', '2016', '2017'],
-        ['Matcha Latte', 43.3, 85.8, 93.7],
-        ['Milk Tea', 83.1, 73.4, 55.1],
-      ];
-      =>
-      [
-        ['product', 'Matcha Latte', 'Milk Tea'],
-        ['2015', 43.3, 83.1],
-        ['2016', 85.8, 73.4],
-        ['2017', 93.7, 55.1],
-      ];
-     */
-    validate(source);
-
-    const newSource = source[0].map((col, i) => source.map(row => row[i]));
+    const { source} = this.props;
+   
+
+    var key = _.map(source, _.keys)[0];
```

Comment:
Do not use "var".

Please use "const" and add some meaningful names for each "steps"